include/remill/Arch/Arch.h

```diff
@@ -281,14 +285,16 @@ class Arch {
   //            walk up, one byte at a time, to `MaxInstructionSize(false)`
   //            bytes being passed to the decoder, until you successfully decode
   //            or ultimately fail.
-  virtual bool DecodeInstruction(uint64_t address, std::string_view instr_bytes,
-                                 Instruction &inst) const = 0;
+  virtual std::optional<DecodingContext::ContextMap>
+  DecodeInstruction(uint64_t address, std::string_view instr_bytes,
+                    Instruction &inst, DecodingContext context) const = 0;
```

Comment:
`context` is taken by value - is that an intentional choice and we expect to always take care of ownership? Or do we expect to mutate it? (Just trying to understand if `const &` is not a better choice).